src/main/java/io/debezium/connector/yugabytedb/connection/YBArray.java

```diff
@@ -0,0 +1,180 @@
+package io.debezium.connector.yugabytedb.connection;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.postgresql.core.BaseConnection;
+import org.postgresql.core.Parser;
+//import org.postgresql.jdbc.ArrayDecoding;
+//import org.postgresql.jdbc.ArrayEncoding;
+import org.postgresql.jdbc.PgArray;
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class YBArray extends PgArray {
```

Comment:
Are there unit tests that we can also bring in especially for functions like `buildArrayList`